integration/tests/identity_provider/identity_provider_test.go

```diff
@@ -0,0 +1,379 @@
+	if err != nil {
+		return nil, errors.Wrap(err, "unexpected error decoding certificate authority data")
+	}
+
+	config := clientcmdapi.Config{
+		Clusters: map[string]*clientcmdapi.Cluster{
+			clusterName: {
+				Server:                   *cluster.Cluster.Endpoint,
+				CertificateAuthorityData: certData,
+			},
+		},
+		Contexts: map[string]*clientcmdapi.Context{
+			contextName: {
+				Cluster:  clusterName,
+				AuthInfo: contextName,
+			},
+		},
+		AuthInfos: map[string]*clientcmdapi.AuthInfo{
+			contextName: {
+				AuthProvider: &clientcmdapi.AuthProviderConfig{
+					Name: "oidc",
+					Config: map[string]string{
+						"client-id":      o.clientID,
+						"id-token":       o.idToken,
+						"refresh-token":  o.refreshToken,
+						"idp-issuer-url": o.idpIssuerURL,
+					},
+				},
+			},
+		},
+		CurrentContext: contextName,
+	}
+
+	clientConfig, err := clientcmd.NewDefaultClientConfig(config, &clientcmd.ConfigOverrides{}).ClientConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "creating default client config")
+	}
+
+	return kubernetes.NewForConfig(clientConfig)
+}
+
+func setupCognitoProvider(clusterName, region string) (*OIDCConfig, error) {
+	c := cognitoidentityprovider.New(NewSession(region))
+
+	pool, err := c.CreateUserPool(&cognitoidentityprovider.CreateUserPoolInput{
+		Policies: &cognitoidentityprovider.UserPoolPolicyType{
+			PasswordPolicy: &cognitoidentityprovider.PasswordPolicyType{
+				MinimumLength:    aws.Int64(10),
+				RequireLowercase: aws.Bool(false),
+				RequireNumbers:   aws.Bool(true),
+				RequireSymbols:   aws.Bool(true),
+				RequireUppercase: aws.Bool(false),
+			},
+		},
+		PoolName:           aws.String(clusterName),
+		UsernameAttributes: aws.StringSlice([]string{"email"}),
+	})
+
+	if err != nil {
+		return nil, errors.Wrap(err, "creating user pool")
+	}
+
+	cleanupCognitoResources = func() error {
+		_, err := c.DeleteUserPool(&cognitoidentityprovider.DeleteUserPoolInput{
+			UserPoolId: pool.UserPool.Id,
+		})
+		return err
+	}
+
+	client, err := c.CreateUserPoolClient(&cognitoidentityprovider.CreateUserPoolClientInput{
+		ClientName: aws.String("eks-client"),
+		ExplicitAuthFlows: aws.StringSlice([]string{
+			cognitoidentityprovider.ExplicitAuthFlowsTypeAllowAdminUserPasswordAuth,
+			cognitoidentityprovider.ExplicitAuthFlowsTypeAllowUserPasswordAuth,
+			cognitoidentityprovider.ExplicitAuthFlowsTypeAllowRefreshTokenAuth,
+		}),
+		AllowedOAuthFlows: aws.StringSlice([]string{
+			cognitoidentityprovider.OAuthFlowTypeImplicit,
+		}),
+		AllowedOAuthScopes: aws.StringSlice([]string{
+			"profile",
+			"phone",
+			"email",
+			"openid",
+			"aws.cognito.signin.user.admin",
+		}),
+		UserPoolId:                 pool.UserPool.Id,
+		GenerateSecret:             aws.Bool(false),
+		SupportedIdentityProviders: aws.StringSlice([]string{"COGNITO"}),
+		// TODO this is likely not required, check if this can be removed.
```

Comment:
I am going to open another issue for this, as it requires updating values for other fields.